### ISSUE TITLE
Enable declaration maps for hyperclick-to-implementation

### ISSUE THREAD

**Issue by steveluscher** [solana-labs/solana-web3.js#3576]
## Motivation

Right now if you hyperclick on a symbol in VS Code it will take you to the typedef. If you want to go to the _implementation_ you have to right click and select ‘Go to Implementations.’

Apparently, if you generate declaration maps (like sourcemaps, for typedefs) then hyperclick goes to the implementation instead.

## Details

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---declarationmap

**Comment by shivaji43** [comment]
what do we have to do exactly , can you please explain a bit , because the declarationMap changes are already there in the tsconfig.json file , which generate the maps.

**Comment by steveluscher** [comment]
You're not wrong, @shivaji43! I was hoping that posting this issue would mean that I could stop thinking about this problem, but here we are :)

Having looked into it, it dawns on me that hyperclick (go to definition) does not work today because we don't include the source code in the published npm packages.

I _really_ didn't want to include the source in the published npm packages, because this causes strife for people who – for instance – publish their software to cloud services like AWS Lambda, and don't want to publish megabytes of code that does nothing.

So I guess we have a few options.

* Forget it.
* Include source files in the npm packages (by adding them to the `files` field of package.json) package size be damned.
* Generate declaration maps from the _compiled_ source code (ie. the files in `dist/*.mjs`).
* A fourth option I haven't thought of?

**Comment by shivaji43** [comment]
Including source files is not a good idea , I think even generating declaration maps can take significant size in the packages , as far as I know it is not possible without the source files , if you can suggest me some more options (if there) I am open to do them.
